### ISSUE TITLE
Regressions from #5675

### ISSUE THREAD

**Issue by kd-11** [RPCS3/rpcs3#5708]
A regression tracker for any remaining bugs introduced by https://github.com/RPCS3/rpcs3/pull/5675.

EDIT: When reporting a regression, check if its fixed by enabling WCB and/or forcing CPU blit from the debug option.

TODO:
- [x] JAK II & III
- [x] Backbreaker
- [x] Rogue Legacy
- [x] Iron Man
- [x] MCLA
- [x] RE6
- [x] Genji
- [x] Yakuza 4
- [x] Silent Hill: Downpour
- [x] MS: Apocalypse
- [x] Ben 10 Galactic Racing
- [x] Monster Jam
- [x] Bioshock

**Comment by MSuih** [comment]
Yakuza 4 Ingame is now almost completely black now, only showing some lights in distance. Also there seems to be a still image that constantly flickers over the right side of the screen. Affects both opengl and vulkan.
![image](https://user-images.githubusercontent.com/19759300/54086936-db7a8900-4356-11e9-91bd-eb7df60e66c5.png)

The game also spams `RSX: Blit request consists of an empty region descriptor!` to log.

Indoor segments look a bit better, sometimes showing half or even full frame of correct graphics through the flicker.
![image](https://user-images.githubusercontent.com/19759300/54086695-89d0ff00-4354-11e9-9caa-9ac935bde85e.png)

Enabling WCB or CPU blit separately don't change anything. Enabling them both on OpenGL prints the following error to log and crashes the whole emulator: `RSX: g_access_violation_handler(0xc2650004, 0): Assertion failed: src != nullptr
(in file ...\rpcs3\emu\rsx\common\texture_cache_utils.h:1403)`
On vulkan it fixes the flicker, but graphics are still mostly black.
![image](https://user-images.githubusercontent.com/19759300/54087046-e4b82580-4357-11e9-9f93-59f44f809cf7.png)

In an earlier version of the PR Yakuza 4 worked fine, but I wasn't checking it often enough to know which change caused it to regress.

**Comment by legend800** [comment]
Do you not want issues that are fine with WCB + Blit?

Regression List:

**Burnout Crash**: (with or without WCB + Blit)
No longer playable - mad flickering of objects in levels following by a crash within a min or so (2/2 repro). GFX/stability perfect on Master. Happens with/w/o WCB+Blit.

S {RSX Decompiler Thread} RSX: New program compiled successfully x6
F {RSX [0x025d690]} RSX: class std::runtime_error thrown: Verification failed:
(in file c:\projects\rpcs3\rpcs3\emu\rsx\common\texture_cache.h:2101)

**Backbreaker Vengeance:**
Broken "Touchdown" text (only with WCB+Blit off):
![bb2](https://user-images.githubusercontent.com/20983844/54087663-4d7caf00-4312-11e9-9769-865d780f0f7a.png)

Broken field logo: (with or without WCB + Blit)
![bb](https://user-images.githubusercontent.com/20983844/54087661-4655a100-4312-11e9-8d77-4fb7806409f6.png)

**Motorstorm Apoc.** - (Needs wcb to render ingame gfx, blit makes everything black)
Does a weird thing now, where it shows a discolored edge that moves itself to the edge of screen and makes the black border bigger than normal. Strange.

Master:
![moto1](https://user-images.githubusercontent.com/20983844/54087582-80727300-4311-11e9-86cd-f29582786ad2.png)

PR:
![moto2](https://user-images.githubusercontent.com/20983844/54087589-8cf6cb80-4311-11e9-9723-cf8f84b5ffca.png)

Then transitions to this....
![moto3](https://user-images.githubusercontent.com/20983844/54087592-8ff1bc00-4311-11e9-89ae-93f6d959afe4.png)

**Comment by jenci8888** [comment]
Elad's PR merged from master branch, I've got excited to test GT6 game

Game: GT6
Settings: All tried, no changes effect.
Vulkan: `F {RSX [0x119dc0c]} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vkhelpers.h:1278)`
OGL: `F {RSX [0x1918d68]} RSX: class std::runtime_error thrown: Assertion failed: src != nullptr
(in file c:\projects\rpcs3\rpcs3\emu\rsx\common\texture_cache_utils.h:1403)`

Game: GT5
Settings: Default
Vulkan:  `F {RSX [0x0013080]} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vkhelpers.h:1278)` (night tracks including wet/time changeable)

**Comment by Kravickas** [comment]
**Far Cry 3**

Parts of the menu are corrupted
![obrazek](https://user-images.githubusercontent.com/29986095/54090215-00302a00-4372-11e9-94b2-b12497d0b95f.png)

I have RSX capture - but in few tries it didnt show up there and OGL is stuck before any splash screen and in rsx capture it is fine as well.
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2949948/RPCS3.log.gz)

No longer spam of `U {RSX [0x00836fc]} RSX TODO: Resource used before memory initialization` but instead it spam few time `E 0:00:38.630530 {RSX [0x00721ac]} RSX: Unsupported depth conversion (0x9C)`

**Comment by uaqlover** [comment]
issue with Ben 10™ Galactic Racing [BLUS30822]

![12333](https://user-images.githubusercontent.com/45731484/54092414-f6460100-4348-11e9-80f9-92f508a0e673.png)


![RPCS3_ Fatal Error 10_03_2019 3_24_55 PM](https://user-images.githubusercontent.com/45731484/54092412-ed552f80-4348-11e9-917c-1a0a03e7035a.png)

log file 
[RPCS3 10.log.gz](https://github.com/RPCS3/rpcs3/files/2950129/RPCS3.10.log.gz)

**Comment by uaqlover** [comment]
rpcs3-v0.0.6-7857-1875dc3f_win64

NASCAR® Unleashed [BLUS30806]

after playing like 5 minute it freeze 

video 

https://drive.google.com/file/d/1it4278clFvGgjMzk3Cx_yPaSb22fXxFG/view?usp=sharing

![FPS_ 321 76 _ OpenGL _ 0 0 6-7857 _ NASCAR® Unleashed  BLUS30806  10_03_2019 3_37_10 PM](https://user-images.githubusercontent.com/45731484/54092524-6c973300-434a-11e9-96e8-3558a85bb059.png)


F {RSX [0x0061b9c]} RSX: class std::runtime_error thrown: Assertion failed: src != nullptr
(in file c:\projects\rpcs3\rpcs3\emu\rsx\common\texture_cache_utils.h:1403)

Rampage World Tour [NPUB30003]

is issue is back on master

F {PPU[0x1000006] Thread (Thread 1) [0x0013a3cc]} MEM: Access violation reading location 0x0
W {cellAudio Thread} cellAudio: Audio backend stopped unexpectedly, likely due to a buffer underrun

but when ever i try rpcs3-v0.0.6-49120261_win64 is working

![FPS_ 59 96 _ Vulkan _ 0 0 6-49120261-fbo_fixes _ Rampage World Tour  NPUB30003  10_03_2019 3_43_51 PM](https://user-images.githubusercontent.com/45731484/54092627-93a23480-434b-11e9-95c2-080db632ef2f.png)

**Comment by BreakPoints** [comment]
Emulation freezes with:

`F {RSX [0x012afe8]} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\gltexturecache.h:200)`

when launching Jak 2 or 3 from Jak and Daxter Collection under both Vulkan and OpenGL

**Comment by uaqlover** [comment]
Monster Jam: Path of Destruction [BLUS30561]

(cpu Blit off):

![FPS_ 29 58 _ Vulkan _ 0 0 6-7857 _ Monster Jam_ Path of Destruction  BLUS30561  10_03_2019 7_27_59 PM](https://user-images.githubusercontent.com/45731484/54096895-08389b80-436b-11e9-91e9-0808b802cd4d.png)

(cpu Blit on):

![FPS_ 17 31 _ Vulkan _ 0 0 6-7857 _ Monster Jam_ Path of Destruction  BLUS30561  10_03_2019 7_29_56 PM](https://user-images.githubusercontent.com/45731484/54096907-1c7c9880-436b-11e9-876b-5ca122375707.png)

**Comment by alexloxldlol** [comment]
![image](https://user-images.githubusercontent.com/17052352/54132588-c82df300-4414-11e9-98b5-ef54289d931a.png)

**Comment by Shoegzer** [comment]
Bioshock also seems to be affected, as noted in #5609:

![pic](https://user-images.githubusercontent.com/12475371/54167048-e240f300-445f-11e9-860d-51f36cfe1706.png)

**Comment by stride21** [comment]
Previous Master:
![ratchet master](https://user-images.githubusercontent.com/37915864/54221642-ef8ec980-44c1-11e9-88fb-46331ac0d0f0.JPG)
Master:
![Capture](https://user-images.githubusercontent.com/37915864/54221649-f3225080-44c1-11e9-82ce-00301cac357e.JPG)
Vulkan:
![vulkan resis](https://user-images.githubusercontent.com/37915864/54221910-5ca25f00-44c2-11e9-8890-7c00cb496173.JPG)
OpenGL:
![rpcs3 resis](https://user-images.githubusercontent.com/37915864/54221657-f87f9b00-44c1-11e9-878b-42cfb07e7dec.JPG)

**Comment by SkyEua** [comment]
In this new version (rpcs3-v0.0.6-7867-980c3d6f_win64).

![001](https://user-images.githubusercontent.com/15639365/54297661-0ef72680-4596-11e9-840f-8c393e05f9dc.PNG)

The menu still continues with the same problem; but in the game.
The game came back to work normally, only in these parts of "balance" happens an FPS fall

![002](https://user-images.githubusercontent.com/15639365/54297749-32ba6c80-4596-11e9-97ce-71a67b4e3755.PNG)

What did not happen in this version of the RPCS3 emulator v0.0.6-7840-a43e7c17, even running in 5K, was perfect

Note: from the RPCS3 version v0.0.6-7840-a43e7c17 - FOR - RPCS3 v0.0.6-7857-1875dc3f the game gets injurious 0,73FPS

**Comment by kd-11** [comment]
Need some general feedback with this build https://ci.appveyor.com/project/kd-11/rpcs3/builds/23040747/artifacts.
Note: Please check both OpenGL and Vulkan and note differences if any. Flickering with vulkan in some games is expected due to uneven performance desynchronizing Cell threads. This is being worked on. If it works fine on OpenGL and is mysteriously flickering with vulkan that is likely the cause.

**Comment by BreakPoints** [comment]
The regression that reported has been fixed in the new build. Not noticing any differences with master in Jak and Daxter's case.

> Emulation freezes with:
> 
> `F {RSX [0x012afe8]} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): (in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\gltexturecache.h:200)`
> 
> when launching Jak 2 or 3 from Jak and Daxter Collection under both Vulkan and OpenGL

**Comment by uaqlover** [comment]
working with this build 

Rampage World Tour  NPUB30003  

![FPS_ 63 94 _ Vulkan _ 0 0 6-7862 _ Rampage World Tour  NPUB30003  13_03_2019 2_37_55 PM](https://user-images.githubusercontent.com/45731484/54322557-dc0f5b80-45b2-11e9-9e84-ff87daf0278e.png)

 Monster Jam: Path of Destruction [BLUS30561] is same 

![FPS_ 29 77 _ Vulkan _ 0 0 6-7862 _ Monster Jam_ Path of Destruction  BLUS30561  13_03_2019 2_37_22 PM](https://user-images.githubusercontent.com/45731484/54322588-f77a6680-45b2-11e9-9eb9-b253a8b3107f.png)

**Comment by stride21** [comment]
In Resistance 3 the size of the screen portion that shows graphics when using the auger sight can expand and contract or completely disappear. When it contracts it can leave a still image that is the same size as the previous portion and it doesn't go away when it happens but with the PR it no longer shows still images when using the auger.
![resistance 3 no still](https://user-images.githubusercontent.com/37915864/54325531-c0b34900-45d0-11e9-8bdc-da95a63257cf.JPG)

Ratchet and Clank Future: A Crank in Time is still the same with the recent PR.

**Comment by SkyEua** [comment]
RPCS3 v0.0.6-7862-154c8a2e
With vulkan, the game presents itself correctly, all the time in 30FPS even in 4k. 
Only the opening screen, below 9FPS this time.
But occasionally I get the error:
`F {RSX [0x014dea8]} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vkhelpers.h:1278)`

![image](https://user-images.githubusercontent.com/15639365/54325719-789d2200-45e2-11e9-98be-4cb2c8669b87.png)


Already in OpenGL, It behaves the same way, low fps only in opening; running 30FPS at all times.
But when it comes to the "balance" part the FPS drops.

![image](https://user-images.githubusercontent.com/15639365/54326537-6a9cd080-45e5-11e9-91d3-4e717705b4fb.png)

I would have sent the log file, but it gets really big 2GB ????

**Comment by legend800** [comment]
Results from new build:

**No change:**
Backbreaker TD text/logo (vengeance only)

**Fixed:**
Burnout flicker/crash

**Worse:**
Motorstorm Apocalypse border issue:
![2019-03-13 19_56_51-FPS_ 12 76 _ Vulkan _ 0 0 6-7862 _ MOTORSTORM™_ APOCALYPSE Digital Edition  NPUA](https://user-images.githubusercontent.com/20983844/54328534-7da2a700-45cb-11e9-869d-4b24beab4bc0.png)

**Comment by jenci8888** [comment]
GT5 and GT6 fails at intro, when SPU race condition breaks, both same error in log. (I've tried change all settings and no effect.)
However, Master doesn't have a problem, either. (Still remains RSX verification error)
`F {SPU[0x2000001] Thread (PDICellSpursKernel1) [0x08074]} MEM: Access violation writing location 0x611e0000`
Log: [RPCS3.zip](https://github.com/RPCS3/rpcs3/files/2966192/RPCS3.zip)

**Comment by kd-11** [comment]
https://ci.appveyor.com/project/kd-11/rpcs3/builds/23070933 Should fix unstable vulkan image (flickering)

**Comment by kd-11** [comment]
@jenci8888 I can't look into it if it dies with SPU crash; I certainly do not touch anything on SPU with this change.

**Comment by elad335** [comment]
This looks like a page fault miss.

**Comment by MarioSonic2987** [comment]
MCLA now shows pink flickering reflections. Enabling Force CPU blit doesn't fix it.

**Before #5675**:
![image](https://user-images.githubusercontent.com/20027681/54365201-b0ea4200-466e-11e9-99ae-da619c5b61a2.png)

**#5675 and after**:
![image](https://user-images.githubusercontent.com/20027681/54364820-f2c6b880-466d-11e9-81fc-93d4ad14d5a1.png)

I've tested with the build provided by kd-11 and it still has this bug.

**Comment by kd-11** [comment]
You forgot to clarify what OpenGL looks like or if strict mode does anything.

**Comment by Emulator-Team-2** [comment]
@jenci8888 
The fbo_fixes branch is not rebased on the latest master, so there is no fix for gt 5/6 in this branch.

**Comment by MSuih** [comment]
The only change for Yakuza 4 is that the OpenGL crash has changed to `RSX: class std::runtime_error thrown: Verification failed: 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\glhelpers.h:303)`

**Comment by MarioSonic2987** [comment]
<details>
<summary>MCLA OpenGL</summary>

**Before #5675**:
With WCB:
![image](https://user-images.githubusercontent.com/20027681/54368566-b185d700-4674-11e9-9d8f-574892924d1a.png)
Without WCB:
![image](https://user-images.githubusercontent.com/20027681/54369114-a5e6e000-4675-11e9-975b-6c1debc82030.png)


**#5675 and after**:
Without WCB, has black flickering reflections.
![image](https://user-images.githubusercontent.com/20027681/54368853-2eb14c00-4675-11e9-9c58-48780f3dd480.png)

With WCB, throws this in console and crashes the emulator:
```
·F 0:01:26.276390 {RSX [0x001e01c]} RSX: class std::runtime_error thrown: Assertion failed: src != nullptr
(in file c:\projects\rpcs3\rpcs3\emu\rsx\common\texture_cache_utils.h:1403)
```
With fbo_fixes new build, it still has black reflections without WCB. CPU blit doesn't fix it.
With WCB, throws this in console and crashes the emulator:
```
F {RSX [0x0031afc]} RSX: class std::runtime_error thrown: Verification failed: 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\glhelpers.h:303)
```
</details>

**Comment by SkyEua** [comment]
> https://ci.appveyor.com/project/kd-11/rpcs3/builds/23070933 Should fix unstable vulkan image (flickering)

In the "balance" part this is working normally, even in the part of the introduction.
Only in this part that remains the same.

![image](https://user-images.githubusercontent.com/15639365/54370489-75dc1480-4656-11e9-8926-6501c921bf95.png)

**Comment by kd-11** [comment]
@MSuih So vulkan is still broken? I'll look into the OGL issue.
@SkyEua We had this regression before; I remember it was easy to fix - will look into it.
Fortunately the most dangerous bug (flickering, which means race conditions) is gone now so the rest should be manageable.

**Comment by Emulator-Team-2** [comment]
After #5675 
Iron Man-Demo now hangs up on the loading screen before entering the game.
Master:
Last error in the log:
`RSX: Depth<->RGBA blit on a framebuffer requested but not supported`
`RSX: Cache miss at address 0xC1EB8000. This is gonna hurt...`
Log:
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/2967813/RPCS3.zip)

With WCB + CPU blit crashes with fatal error:
[Vulkan]
`F {RSX [0x0007860} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vkgsrender.cpp:3318)`

[OpenGL] 
`F {RSX [0x00072c0]} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\glgsrender.cpp:1649)`
Log:
[RPCS3-Log-Master.zip](https://github.com/RPCS3/rpcs3/files/2967946/RPCS3-Log-Master.zip)


---------------------------------------------------------------------------------------------------------
With Latest build from [fbo_fixes] branch:

Still hangs on the loading screen, but show more errors:
`RSX: Depth<->RGBA blit on a framebuffer requested but not supported`
`RSX: Area merge failed! addr=0xc1ab8000, w=1024, h=4096, gcm_format=0x85[sz=0]`
`RSX: Btw, you're about to lose a blit surface at 0xc1eb8000`
Log:
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/2967857/RPCS3.zip)

With WCB + CPU blit:
[OpenGL]:
`RSX: g_access_violation_handler(0xc0fe0000, 0): Verification failed: `
`(in file c:\projects\rpcs3\rpcs3\emu\rsx\gl\glhelpers.h:303)`

[Vulkan]:
`F {RSX [0x000750c]} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): `
`(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vkgsrender.cpp:3287)`
Log:
[RPCS3-Log.zip](https://github.com/RPCS3/rpcs3/files/2967923/RPCS3-Log.zip)

**Comment by MSuih** [comment]
@kd-11 Yes, Vulkan is still broken.

**Comment by MsDarkLow** [comment]
Before #5675, Vulkan and OpenGL render correctly without the need of WCB.
![GOOD](https://user-images.githubusercontent.com/9060436/54430099-5c040580-471a-11e9-81e5-b93494e66f40.png)
[RPCS3 Before.log.gz](https://github.com/RPCS3/rpcs3/files/2970887/RPCS3.Before.log.gz)

#5675 and after, Vulkan and OpenGL render incorrectly the background.
![BAD 1](https://user-images.githubusercontent.com/9060436/54430149-85bd2c80-471a-11e9-9595-2c5073582c42.png)
[RPCS3 After Vulkan.log.gz](https://github.com/RPCS3/rpcs3/files/2970923/RPCS3.After.Vulkan.log.gz)


w/ WCB enabled it solves a part of the issue, forcing CPU Blit doesn't help.
![BAD 2](https://user-images.githubusercontent.com/9060436/54430162-9077c180-471a-11e9-8434-eb6485df93c8.png)
[RPCS3 Vulkan WCB.log.gz](https://github.com/RPCS3/rpcs3/files/2970898/RPCS3.Vulkan.WCB.log.gz)

OpenGL + WCB throws an error.
`F {RSX [0x00cb378]} RSX: class std::runtime_error thrown: Assertion failed: src != nullptr
(in file c:\projects\rpcs3\rpcs3\emu\rsx\common\texture_cache_utils.h:1403)`
[RPCS3 OpenGL WCB.log.gz](https://github.com/RPCS3/rpcs3/files/2970901/RPCS3.OpenGL.WCB.log.gz)

**Comment by kd-11** [comment]
New build here https://ci.appveyor.com/project/kd-11/rpcs3/builds/23107321/artifacts. Retest.

**Comment by MSuih** [comment]
Yakuza 4 OpenGL crash is fixed, but the graphical issues described in my first message affect both APIs.

**Comment by kd-11** [comment]
Disregard last build if getting crashes; use this build instead https://ci.appveyor.com/project/kd-11/rpcs3/builds/23108496

**Comment by MsDarkLow** [comment]
> New build here https://ci.appveyor.com/project/kd-11/rpcs3/builds/23107321/artifacts. Retest.

With this build, Rogue Legacy seems to render everything correctly again. Both OpenGL and Vulkan

**Comment by MarioSonic2987** [comment]
MCLA pink reflections bug with Vulkan and black reflection bug and crash with WCB with OpenGL are now fixed.

**Comment by kd-11** [comment]
Thanks for confirming the timeout bug, I'm investigating.

**Comment by kd-11** [comment]
[This build](https://ci.appveyor.com/project/kd-11/rpcs3/builds/23113166) should fix the events timing out with vulkan.

**Comment by MarioSonic2987** [comment]
It no longer spams in console and performance returns to normal.
![image](https://user-images.githubusercontent.com/20027681/54462143-0148c900-476f-11e9-941b-3f0277b6dfef.png)

**Comment by MarioSonic2987** [comment]
Worms Crazy Golf and Worms Revolution with Vulkan crash after starting up with this line thrown in console:
```
F {RSX [0x0102f34]} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vktexturecache.h:300)
```
With OpenGL, it's fine.

[Log WCG VK Broken](https://github.com/RPCS3/rpcs3/files/2973003/RPCS3.log.gz)
[Log WCG GL](https://github.com/RPCS3/rpcs3/files/2973013/RPCS3.log.gz)

**Comment by kd-11** [comment]
Should be fixed by https://ci.appveyor.com/project/kd-11/rpcs3/builds/23114644

**Comment by MarioSonic2987** [comment]
Both games still don't work. Throws in console:
```
·F 0:00:13.090633 {RSX [0x0103fb4]} RSX: class std::runtime_error thrown: Verification failed (e=0x7f): 
(in file c:\projects\rpcs3\rpcs3\emu\rsx\vk\vktexturecache.h:310)
```
[Log WCG VK Broken](https://github.com/RPCS3/rpcs3/files/2973109/RPCS3.log.gz)

**Comment by uaqlover** [comment]
rpcs3-v0.0.0.6-7879-5d7b41a3_win64

fix Monster Jam_ Path of Destruction [BLUS30561]

![FPS_ 17 04 _ Vulkan _ 0 0 6-7879 _ Monster Jam_ Path of Destruction  BLUS30561  15_03_2019 3_36_07 PM](https://user-images.githubusercontent.com/45731484/54465434-6208df00-4738-11e9-9c0a-e0fba38ae312.png)


but it show yellow when i stop the car 

![FPS_ 17 04 _ Vulkan _ 0 0 6-7879 _ Monster Jam_ Path of Destruction  BLUS30561  15_03_2019 3_36_48 PM](https://user-images.githubusercontent.com/45731484/54465403-3980e500-4738-11e9-92ea-d5095acbf77a.png)

**Comment by uaqlover** [comment]
also it's stable now NASCAR® Unleashed [BLUS30806] not crashing 


and Surf's Up [BLES00083] broke and flickering again 

![FPS_ 29 96 _ Vulkan _ 0 0 6-7879 _ Surf's Up  BLES00083  15_03_2019 3_56_12 PM](https://user-images.githubusercontent.com/45731484/54466257-37b92080-473c-11e9-9319-029dcb83a753.png)

![FPS_ 29 96 _ Vulkan _ 0 0 6-7879 _ Surf's Up  BLES00083  15_03_2019 3_57_32 PM](https://user-images.githubusercontent.com/45731484/54466263-3be53e00-473c-11e9-804f-83346e0f950c.png)

master

![FPS_ 29 98 _ Vulkan _ 0 0 6-7869 _ Surf's Up  BLES00083  15_03_2019 4_03_37 PM](https://user-images.githubusercontent.com/45731484/54466272-41428880-473c-11e9-9487-94aeb4d588c6.png)

![FPS_ 29 98 _ Vulkan _ 0 0 6-7869 _ Surf's Up  BLES00083  15_03_2019 4_03_57 PM](https://user-images.githubusercontent.com/45731484/54466276-443d7900-473c-11e9-8960-45b75bc18985.png)

**Comment by uaqlover** [comment]
Ben 10™ Galactic Racing [BLUS30822]

error is gone but still has black textures ingame 

![FPS_ 29 96 _ OpenGL _ 0 0 6-7879 _ Ben 10™ Galactic Racing  BLUS30822  15_03_2019 4_11_55 PM](https://user-images.githubusercontent.com/45731484/54466520-523fc980-473d-11e9-9fd3-32a59c8f5e23.png)

![FPS_ 29 96 _ OpenGL _ 0 0 6-7879 _ Ben 10™ Galactic Racing  BLUS30822  15_03_2019 4_11_57 PM](https://user-images.githubusercontent.com/45731484/54466524-59ff6e00-473d-11e9-9d36-19b81e733a99.png)

same is rio has black textures

**Comment by uaqlover** [comment]
Kung Fu Panda® 2 [BLUS30634]

is flickering 

![FPS_ 34 16 _ Vulkan _ 0 0 6-7879 _ Kung Fu Panda® 2  BLUS30634  15_03_2019 4_19_44 PM](https://user-images.githubusercontent.com/45731484/54466774-9e3f3e00-473e-11e9-93a3-917b5791fb78.png)

video 

https://drive.google.com/file/d/1QysoyxmtcfUF0gKTEKen7SWy83Mvgson/view?usp=sharing

**Comment by uaqlover** [comment]
Skydive: Proximity Flight [NPUB30618] is broke 

![FPS_ 333 67 _ Vulkan _ 0 0 6-7879 _ Skydive_ Proximity Flight  NPUB30618  15_03_2019 4_29_20 PM](https://user-images.githubusercontent.com/45731484/54466955-8e742980-473f-11e9-9468-c26fae99bb9d.png)

Shoot Many Robots (Demo) [NPUB30600] is still dark

![FPS_ 30 03 _ Vulkan _ 0 0 6-7879 _ Shoot Many Robots (Demo)  NPUB30600  15_03_2019 4_55_26 PM](https://user-images.githubusercontent.com/45731484/54467608-7e5e4900-4743-11e9-998d-416d02f214b2.png)

**Comment by Emulator-Team-2** [comment]
@uaqlover 
Kung Fu Panda 2 requires to enable "Disable Vertex Cache" to fix this issue.
Stop spamming with unrelated issues.

**Comment by uaqlover** [comment]
thanks @Emulator-Team-2 and sorry for posting unrelated issues

**Comment by uaqlover** [comment]
this pr fix's all pes series games

pr

![FPS_ 60 11 _ Vulkan _ 0 0 6-7879 _ Pro Evolution Soccer 2010  BLUS30404  15_03_2019 6_24_19 PM](https://user-images.githubusercontent.com/45731484/54469049-4f9a9f80-4750-11e9-8cfc-ef7443d62eb6.png)

master

![FPS_ 59 93 _ Vulkan _ 0 0 6-7869 _ Pro Evolution Soccer 2010  BLUS30404  15_03_2019 6_29_13 PM](https://user-images.githubusercontent.com/45731484/54469052-5aedcb00-4750-11e9-8e32-f7cf515ad97b.png)

**Comment by legend800** [comment]
@kd-11 Not only did you fix the field logo regression in Backbreaker, but you fixed this old VK regression making Backbreaker Vengeance, for the first time ever, fully playable with full speed!!
https://github.com/RPCS3/rpcs3/issues/5708
I no longer need to bug you on changes to this game in every PR after all these years. :P 

Finish him!
![2019-03-15 20_26_49-FPS_ 30 01 _ Vulkan _ 0 0 6-7879 _ BACKBREAKER VENGEANCE  NPEB00592](https://user-images.githubusercontent.com/20983844/54470455-55e74680-4765-11e9-9407-6900dbaab609.png)

Motorstorm Apocalypse better, but still has a little bit of a regression:
VK (broken borders on intro to stage, gets a little better after race starts)
![2019-03-15 20_29_50-FPS_ 11 72 _ Vulkan _ 0 0 6-7879 _ MOTORSTORM™_ APOCALYPSE Digital Edition  NPUA](https://user-images.githubusercontent.com/20983844/54470489-a363b380-4765-11e9-8e26-f22e48d95616.png)

OpenGL (broken borders on intro to stage, looks fine after race starts)
![2019-03-15 20_33_55-FPS_ 8 70 _ OpenGL _ 0 0 6-7879 _ MOTORSTORM™_ APOCALYPSE Digital Edition  NPUA8](https://user-images.githubusercontent.com/20983844/54470488-a068c300-4765-11e9-9069-1236a705f15f.png)

**Comment by Xcedf** [comment]
Flickering in GoW 3 finally fixed on GPU Blit, but still left a performance regression from https://github.com/RPCS3/rpcs3/commit/310c4f0feea6d254c8adbaf614e6143e39a4f961

**Comment by uaqlover** [comment]
Regression NASCAR The Game: Inside Line [BLUS30932]

pr

![FPS_ 30 04 _ Vulkan _ 0 0 6-7879 _ NASCAR The Game_ Inside Line  BLUS30932  15_03_2019 11_29_25 PM](https://user-images.githubusercontent.com/45731484/54471704-6d7cfa00-477a-11e9-8d3e-3b764d69297d.png)

master

![FPS_ 30 04 _ Vulkan _ 0 0 6-7869 _ NASCAR The Game_ Inside Line  BLUS30932  15_03_2019 11_28_07 PM](https://user-images.githubusercontent.com/45731484/54471707-74a40800-477a-11e9-97ab-1c0564d5c22b.png)

**Comment by kd-11** [comment]
@Xcedf Unfortunately there is no way around that, the old method just looks up a value and doesn't even check if its valid. It 'just works' for some games but clearly breaks many others.
Optimizations to the new cache mayb alleviate the performance hit, but for now the performance regression has to be there.

**Comment by uaqlover** [comment]
last build from https://github.com/kd-11/rpcs3/commit/fb757cef2fcafc03bcc1d92cab657ac8e7da12bf

Regression NASCAR The Game: Inside Line [BLUS30932] is gone

![FPS_ 29 95 _ Vulkan _ 0 0 6-7879 _ NASCAR The Game_ Inside Line  BLUS30932  16_03_2019 2_42_18 AM](https://user-images.githubusercontent.com/45731484/54473506-fa817c80-4795-11e9-8be9-d3f6f15f51c7.png)

Monster Jam Path of Destruction  BLUS30561 Regression is back

![FPS_ 29 64 _ Vulkan _ 0 0 6-7879 _ Monster Jam_ Path of Destruction  BLUS30561  16_03_2019 2_43_27 AM](https://user-images.githubusercontent.com/45731484/54473509-02d9b780-4796-11e9-88f6-5de1bad37f5a.png)

still has Regression Surf's Up [BLES00083]

![FPS_ 29 92 _ Vulkan _ 0 0 6-7879 _ Surf's Up  BLES00083  16_03_2019 2_46_06 AM](https://user-images.githubusercontent.com/45731484/54473519-213fb300-4796-11e9-9716-431f0020f8df.png)

Ben 10™ Galactic Racing [BLUS30822]
and rio
still has black textures

**Comment by kd-11** [comment]
New build https://ci.appveyor.com/project/kd-11/rpcs3/builds/23121862.
Pre-existing bugs like "black textures" are not getting fixed right now, that is not the point of this work. Its to address regressions compared to previous master versions.
Implemented GPU accelerated rotations which should fix surfs up and Genji regressions.

**Comment by kd-11** [comment]
Last build failed to compiled, fixed here https://ci.appveyor.com/project/kd-11/rpcs3/builds/23122205

**Comment by MarioSonic2987** [comment]
WCG and Worms Revolution still crash, but it's solved with Strict Rendering Mode disabled. Enabling it, crashes both games.

**Comment by kd-11** [comment]
PR opened to address this ticket https://github.com/RPCS3/rpcs3/pull/5726

**Comment by Xcedf** [comment]
Uncharted 2, has vk driver error, and it seems it was here since the beginning of https://github.com/RPCS3/rpcs3/pull/5675, yet the mos of these isues were fixed, this one still here
E {SPU[0x2000003] Thread (highCellSpursKernel3) [0x02238]} RSX: [vulkan] vk::wait_for_event has timed out!
F {RSX [0x7001a20]} RSX: class std::runtime_error thrown: Assertion Failed! Vulkan API call failed with unrecoverable error: Device lost (Driver crashed with unspecified error or stopped responding and recovered) (VK_ERROR_DEVICE_LOST)
(in file c:\users\zero\desktop\rpcs3\rpcs3\emu\rsx\vk\vkhelpers.h:1297)

**Comment by danc2** [comment]
![image](https://user-images.githubusercontent.com/18262983/54477350-0125f780-485b-11e9-9af2-a7fb28281368.png)
when moving camera around him

**Comment by Xcedf** [comment]
@kd-11 despite of performance regression, there's another bug in 310c4f0, a part of water surface has worng color
This PR
![28](https://user-images.githubusercontent.com/7060586/54478802-30227600-4827-11e9-9cc2-4c7def41f70a.jpg)
This PR +310c4f0 reverted
![27](https://user-images.githubusercontent.com/7060586/54478800-2d278580-4827-11e9-8c4d-e5d03b1536ca.jpg)

**Comment by kd-11** [comment]
Confirmed. I suspect there is a pitch incompatibility issue somewhere, will look into it after merging the current PR.

**Comment by Johndeep** [comment]
After #5726 I was eager to test the improvements on Atelier Rorona Plus (BLES02050). My System configuration is Debian Testing, Intel Graphics 620 HD and Mesa 18.4 or/and Modesetting.

Atelier Rorona Plus
Previously (in this case v0.0.6-7825) had no colour texture in many cases, but it didn't flicker
![Bildschirmfoto_2019-03-17_21-41-21](https://user-images.githubusercontent.com/306953/54497698-26b80d00-48fe-11e9-9a62-a2f49a234379.png)
This PR has correct textures (nice!), but the entire screen animation (not visible in the screenshot) flickers so much, it hurts the eyes (not good!)
![Bildschirmfoto_2019-03-17_21-43-12](https://user-images.githubusercontent.com/306953/54497700-2a4b9400-48fe-11e9-9813-19c6bb37a31e.png)

**Comment by Senninz8** [comment]
Ninja Gaiden Sigma 2 crash <<Buffer not big enough>>

Started occurring with PR# 5675 (build 0.0.6-7857). Cannot replicate with PR #5697 build 0.0.6-7840.

![RPCS3_Buffer_Not_big_Enough](https://user-images.githubusercontent.com/48635547/54497915-fd7b8a80-48c5-11e9-9d7f-97fd79e91658.JPG)

[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/2975691/RPCS3.log.gz)

**Comment by kd-11** [comment]
If you see an error about 'blit request denied because dst region does not fit' in the log, retest with this build https://ci.appveyor.com/project/kd-11/rpcs3/builds/23182024.

**Comment by kd-11** [comment]
Retest with https://github.com/RPCS3/rpcs3/pull/5741

**Comment by arabek** [comment]
With #5741 applied Motorstorm (BCES00006) spams the log window with:
RSX: Could not gather all required slices for cubemap/3d generation
when selecting a vehicle

**Comment by arabek** [comment]
Well, that was fast. The logspam is gone with #5742 applied on top of #5741.

**Comment by MSuih** [comment]
Would it be possible to have similar checks in the case where source region does not fit?

**Comment by kd-11** [comment]
Do you have an example where source does not fit? The fix affects both src and dst, but src allows 'clipping' of inputs for performance reasons. Tbh that is a hack and will be removed down the road.

**Comment by kd-11** [comment]
@SkyEua Is silent hill still broken in menu with current master?

**Comment by Emulator-Team-2** [comment]
@kd-11 
Silent Hill: Downpour is still broken on the title screen.
[Vulkan/OpenGL]
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54686256-20ea4380-4b19-11e9-9e81-96ffea9963ba.png)
[RenderDoc V1.2]
[RDC Capture-OpenGL](https://drive.google.com/file/d/1BFiERzGEJN-HwnB-koWyLt9IUp15FTWB/view?usp=sharing)
[RDC Capture-Vulkan](https://drive.google.com/file/d/1CiS1QBuCyiCbK6GODQDwEjgl6-lf7JgR/view?usp=sharing)

**Comment by kd-11** [comment]
@Emulator-Team-2 I'll need an RSX capture for that as well; something is 'off' with that second blit transfer

**Comment by kd-11** [comment]
Also, can you check if WCB makes the issue go away? In theory it should, although there is still a bug there that needs fixing.

**Comment by Emulator-Team-2** [comment]
@kd-11 
WCB doesn't fix this issue.
[RSX Capture]https://drive.google.com/file/d/1W5DYVoeeZwxbrstaiciG9s13PuHwAfXb/view?usp=sharing)

**Comment by MSuih** [comment]
@kd-11 Uncharted 3 during intro and Lollipop Chainsaw during certain scenes. I was thinking it could help with the AMD hw blit memory leak, but on further inspection it seems unrelated as there is one case where blit causes memory leak with nothing in log about blit request failed.

Also I'm not sure if you need an update but Yakuza 4 is still black even after the last PR merge. Without CPU blit and WCB the right side of screen is still messed up, and there seem to be new long lines on left that I haven't seen before. Also with those two disabled it still spams `Blit request consists of an empty region descriptor` to log.

**Comment by stride21** [comment]
Auger sight in Resistance is working correctly when using OpenGL and CPU blit with #5726 but it still has the same issue when you have low health most or all of the screen will turn black with or without using the auger sight. Vulkan still works correct when using the auger sight and when health is low even with SRM only.
OpenGL with SRM and CPU blit:
![new opengl cpu blit](https://user-images.githubusercontent.com/37915864/54704010-edeb8400-4b07-11e9-98f6-e75fecedb8f4.JPG)
Vulkan with SRM and CPU blit:
![new vulkan cpu blit](https://user-images.githubusercontent.com/37915864/54704029-f80d8280-4b07-11e9-95aa-4c79944c7fab.JPG)
OpenGL with SRM and CPU blit(when health is low):
![new opengl low health](https://user-images.githubusercontent.com/37915864/54704016-f04dde00-4b07-11e9-95f7-81ae335d1b10.JPG)
OpenGL SRM:
![srm only](https://user-images.githubusercontent.com/37915864/54704567-fd1f0180-4b08-11e9-9f3a-d91a0b42e506.JPG)
Vulkan SRM:
![vulkan srm new](https://user-images.githubusercontent.com/37915864/54704540-eed0e580-4b08-11e9-8af0-8b9dcdb7e90f.JPG)

**Comment by kd-11** [comment]
Its hopeless to fix the GL bug without renderdocs.

**Comment by kd-11** [comment]
Retest with https://ci.appveyor.com/project/kd-11/rpcs3/builds/23237878. Keep an eye out for regressions to other already fixed bugs and notify me if any.

**Comment by Emulator-Team-2** [comment]
Silent Hill: Downpour is fixed now.
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54726521-ccc07d00-4b73-11e9-99e6-24eaa20ea0b3.png)

**Comment by legend800** [comment]
Motorstorm Apoc. regression still present on latest build.

**Comment by kd-11** [comment]
New build here https://ci.appveyor.com/project/kd-11/rpcs3/builds/23245074 for those getting errors because of using resolution scaler.

**Comment by kd-11** [comment]
PR here https://github.com/RPCS3/rpcs3/pull/5752

**Comment by kd-11** [comment]
Quick note, when testing make sure you first check if everything is working ok with 100% resolution scaling. Non-native rendering can break things so its worth making sure everything is native first and then report if scaling is what is causing problems.

**Comment by MarioSonic2987** [comment]
Resident Evil 6 has regressed with #5741, this door now has black textures. Enabling CPU blit doesn't fix this bug. In console, spams ingame with:

```
E {RSX [0x01cf428]} RSX: Could not gather all required slices for cubemap/3d generation
```

Before #5741:
![image](https://user-images.githubusercontent.com/20027681/54751642-dd0c4280-4bdb-11e9-8c30-b2fbe27f2aed.png)

#5741:
![image](https://user-images.githubusercontent.com/20027681/54751615-c9f97280-4bdb-11e9-8ff8-f1977ce17398.png)

**Comment by kd-11** [comment]
Don't compare to an old PR if a newer one is available.

**Comment by MarioSonic2987** [comment]
With #5752, it still has black textures:
![image](https://user-images.githubusercontent.com/20027681/54756438-848f7200-4be8-11e9-9a89-4ebabb9ce332.png)

**Comment by Emulator-Team-2** [comment]
@kd-11 
RDC and RSX Capture for Monster Jam: Path of Destruction.
[Vulkan]
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54762739-f53c8b80-4bf4-11e9-8413-9d7a1e405b3b.png)
[OpenGL]
![Bez tytułu1](https://user-images.githubusercontent.com/38318848/54762770-04bbd480-4bf5-11e9-886d-97bd577ea286.png)
[RenderDoc V1.2]
[RDC Capture-Vulkan](https://drive.google.com/file/d/1ybv3ueUIeThXoXbe-7Mati1e-EXE2jtS/view?usp=sharing)
[RDC Capture-OpenGL](https://drive.google.com/file/d/1dksfOrFLQFhUJ7FOR85lrFZcKvmyydUI/view?usp=sharing)
[RSX Capture-Vulkan](https://drive.google.com/file/d/1yk71kI-45rOB78V0DERghOINfIDqKnAG/view?usp=sharing)
[RSX Capture-OpenGL](https://drive.google.com/file/d/1nVsyS_cHna6wpQbJFdhTyCRGEwjwp4Ba/view?usp=sharing)

**Comment by legend800** [comment]
Motorstorm still has broken borders, even at 100% res.

![2019-03-21 08_25_10-FPS_ 29 95 _ Vulkan _ 0 0 6-41b66c0f-fbo_fixes _ MOTORSTORM™_ APOCALYPSE Digital](https://user-images.githubusercontent.com/20983844/54763654-96a2ed80-4bb3-11e9-8283-a265ee003a2e.png)

RDC:
https://drive.google.com/file/d/1Cby0koN3NFmC-9Omq92Yguw2lI_Hxlew/view?usp=sharing

**Comment by kd-11** [comment]
Pushed an intersection fix to the new PR, you can retry with that.

**Comment by legend800** [comment]
So it's 90% better. It's still off in the menus, pause menu, start of race and cutscenes. But once the race gets going, it's fine. So you're definitely make changes in the right area.

![2019-03-21 11_03_58-Window](https://user-images.githubusercontent.com/20983844/54774616-66664980-4bc9-11e9-8e8a-c41b8f3f224f.png)
![2019-03-21 11_05_05-Window](https://user-images.githubusercontent.com/20983844/54774618-66664980-4bc9-11e9-9c01-5d35a1e471a5.png)

**Comment by MarioSonic2987** [comment]
Black textures in RE6 are now fixed.
![image](https://user-images.githubusercontent.com/20027681/54775092-84559d80-4c0d-11e9-9311-ec3750591a72.png)

**Comment by kd-11** [comment]
@Emulator-Team-2 Is Monster Jam fixed now?

**Comment by Emulator-Team-2** [comment]
@kd-11 
Tested with latest build.
[Vulkan]
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54779624-eca97c80-4c17-11e9-87e4-bf64bfd69eb6.png)
[OpenGL]
![Bez tytułu1](https://user-images.githubusercontent.com/38318848/54779630-f16e3080-4c17-11e9-9e29-4f24fb914521.png)
[RenderDoc V1.2]
[RDC Capture-Vulkan](https://drive.google.com/file/d/1DdosDnsymvvwyI_MjYPHeqU-UXDUhc6v/view?usp=sharing)
[RDC Capture-OpenGL](https://drive.google.com/file/d/1XPEZgaSjqCs-r1WGu5-VCuvljtAIy-Wg/view?usp=sharing)
[RSX Capture-Vulkan](https://drive.google.com/file/d/136e37XMXdDLYeXZlHMzjVIVIBgOpmiEw/view?usp=sharing)
[RSX Capture-OpenGL](https://drive.google.com/file/d/1gTtMEsPygt0iRb328j4xw1Wj-V6kjPxs/view?usp=sharing)

**Comment by vsub** [comment]
Eternal Sonata - the game is running really slow in a lot of places(weak cpu)but for example in Tenuto Village when I just load my save and do nothing for the next 20 seconds,the max speed I can reach is 18fps but since 0.0.6-7857 till 7888,the max speed I can get is 12fps

So 5-6fps drop since #5675

**Comment by kd-11** [comment]
Pushed a fix for Monster Jam issue, retest other affected titles.

@vsub Performance is not a primary focus at the moment. Optimization will come later after accurate rendering is achieved. There are still several PRs left in the series.

**Comment by Emulator-Team-2** [comment]
Monster Jam: Path of Destruction is fixed now on Vulkan and OpenGL thanks.
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54820878-df84a000-4ca0-11e9-8ba5-4386003e1e6b.png)
![Bez tytułu1](https://user-images.githubusercontent.com/38318848/54820881-e1e6fa00-4ca0-11e9-91fd-4c01dbcf438c.png)

**Comment by kd-11** [comment]
I suppose the only graphics issue remaining issue here is now Yakuza 4, right?

**Comment by legend800** [comment]
That and the Motorstorm Apocalypse black screen areas I think.

**Comment by kd-11** [comment]
Did you check @legend800? There were like 3 more commits/changes since your last test.

**Comment by uaqlover** [comment]
Polar Panic (Demo) [NPUB30113]

![FPS_ 21 67 _ Vulkan _ 0 0 6-7901 _ Polar Panic (Demo)  NPUB30113  23_03_2019 4_45_09 AM](https://user-images.githubusercontent.com/45731484/54865725-49478d00-4d27-11e9-9a07-dac9080655ee.png)

Legends [BLES01614]


![FPS_ 60 09 _ Vulkan _ 0 0 6-7901 _ 007™ Legends  BLES01614  23_03_2019 4_47_57 AM](https://user-images.githubusercontent.com/45731484/54865731-52385e80-4d27-11e9-85a2-7ae8c40ef680.png)

![FPS_ 60 09 _ Vulkan _ 0 0 6-7901 _ 007™ Legends  BLES01614  23_03_2019 4_48_58 AM](https://user-images.githubusercontent.com/45731484/54865739-61b7a780-4d27-11e9-9344-439730e63dd7.png)


Trine 2 Demo  NPUB30331

![FPS_ 29 98 _ Vulkan _ 0 0 6-7901 _ Trine 2 Demo  NPUB30331  23_03_2019 4_50_42 AM](https://user-images.githubusercontent.com/45731484/54865732-582e3f80-4d27-11e9-922e-560f1d2ff776.png)

**Comment by Emulator-Team-2** [comment]
@uaqlover 
I don't know why you are doing this, again  you report false regressions unrelated to this issue.
- Polar Panic - is broken even before #5675. - (rsx: Texturing improvements)
- Trine 2 is broken even before #5675.
- 007 Legends probably broken also before #5675.

**Comment by Johndeep** [comment]
About Hyperdimension Neptunia [BLUS30701], after that PR, there was some change happening.
Again, I'm comparing between rpcs3-v0.0.6-7825 and now rpcs3-v0.0.6-7901 (since the 7825 was about the last "stable" before this huge texture change).
Furthermore, I wouldn't say it is a regression, but it seems that the blur effect in on the one hand finally present, BUT on the other hand, it is.. messed up.

After texture improvements, there are blur effects now, but not quite correct. (rpcs3-v0.0.6-7901) (Please don't mind the fps, because I paused the game)
![Bildschirmfoto_2019-03-23_13-49-58](https://user-images.githubusercontent.com/306953/54866426-b84ad300-4d73-11e9-809e-8f24fff7eda5.png)

No blur effects at all. (rpcs3-v0.0.6-7825)
![Bildschirmfoto_2019-03-23_13-51-29](https://user-images.githubusercontent.com/306953/54866430-be40b400-4d73-11e9-86c1-d26f215f799e.png)

**Comment by uaqlover** [comment]
@Emulator-Team-2 i was thinking maybe it could be fixed in this pr because that i post the games : )

**Comment by Emulator-Team-2** [comment]
@uaqlover 
can you read?
A regression tracker for any remaining bugs introduced by #5675.

**Comment by uaqlover** [comment]
@Emulator-Team-2 my bad

**Comment by Emulator-Team-2** [comment]
@kd-11 
Bioshock #5609 is still broken after #5675 without enable wcb + cpu blit.
Turok have the same problem after #5675.
[Vulkan]
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54867641-df5cd100-4d82-11e9-99cf-eac0c044a54e.png)
[OpenGL]
![Bez tytułu1](https://user-images.githubusercontent.com/38318848/54867649-f00d4700-4d82-11e9-962d-5b2cca45cdef.png)
[RDC Capture-Vulkan](https://drive.google.com/file/d/1aOJDvnxmIFX2KYWnVyXsKqDvNsT5DXoA/view?usp=sharing)
[RDC Capture-OpenGL](https://drive.google.com/file/d/1wbluBhYm1K7ikMPRUoEhfm_ln0y2-sh7/view?usp=sharing)
[RSX Capture-Vulkan](https://drive.google.com/file/d/16KATQwC-QuhV9ePacD2TUQrmofsJ9Djq/view?usp=sharing)
[RSX Capture-OpenGL](https://drive.google.com/file/d/1XBrYnnVdf32quOvJ45byfi7NlT13UH0J/view?usp=sharing)

**Comment by legend800** [comment]
@kd-11 Yeah, I was testing on the latest build. 

I found that expanding the "screen area" in-game affects the "black area" bug's behavior in those areas (menus, pause menu, track countdown, start of race). If I go 100% screen size (default is 50%) then all those areas are completely black. 

![2019-03-23 08_11_07-Window](https://user-images.githubusercontent.com/20983844/54867935-d13b9000-4d43-11e9-8003-6c8817ce156d.png)

Default screen area:
![2019-03-23 08_11_48-Window](https://user-images.githubusercontent.com/20983844/54867937-e0224280-4d43-11e9-8d37-644aca642309.png)

Screen area changed to 100%:
![2019-03-23 08_17_36-Window](https://user-images.githubusercontent.com/20983844/54867946-24154780-4d44-11e9-923d-3165e4102167.png)

**Comment by kd-11** [comment]
Likely fixed in my appveyor builds, but also breaks a bunch of other things (still wip)

**Comment by legend800** [comment]
Now it's just all black always, no matter the in-game setting (w/ WCB+Blit). Soft locks during launch unless Blit is forced on. 

![2019-03-23 17_03_12-FPS_ 12 67 _ Vulkan _ 0 0 6-7902 _ MOTORSTORM™_ APOCALYPSE Digital Edition  NPUA](https://user-images.githubusercontent.com/20983844/54873189-8004be00-4d8e-11e9-89f8-2e9fc3434aca.png)

**Comment by stormymff** [comment]
Yakuza 4 is still in black screen with WCB and force blit CPU on.

**Comment by Xcedf** [comment]
GoW3  water colors are refixed with your latest push
![12](https://user-images.githubusercontent.com/7060586/54925229-ec162c00-4f1e-11e9-88c6-46cfb467a5c8.jpg)

**Comment by kd-11** [comment]
@legend800 Check newest push for MS Apocalypse, fixed a few more issues. I think It's still broken if you use full display scaling ingame settings, I'll check it tomorrow.
@stormymff Force cpu blit is a debug option, it can't work properly with usable performance for all cases (specifically writes to render target area are often lost and there is no way around it that does not destroy performance). As long as the default settings work correctly, it's fine.

**Comment by Emulator-Team-2** [comment]
Tested with the latest commit: 
BioShock still have pink graphics, but only with a vulkan without enable WCB+Cpu blit.
![Bez tytułu](https://user-images.githubusercontent.com/38318848/54955927-07992b00-4f4f-11e9-9026-5a8e07c1426e.png)
![Bez tytułu1](https://user-images.githubusercontent.com/38318848/54955932-08ca5800-4f4f-11e9-9e04-80d734bb5a62.png)
[RDC Capture-Vulkan](https://drive.google.com/file/d/1Kuso6DaFHBEpe6_EH0aTBABJcOJH9Rx5/view?usp=sharing)
[RDC Capture-OpenGL](https://drive.google.com/file/d/12m4IR37ni4p0-oGzpPy2pVc17XXviFEE/view?usp=sharing)
[RSX Capture-Vulkan](https://drive.google.com/file/d/1rFnilpC3zFnHIQBKRTxPsarEN-0ctkij/view?usp=sharing)
[RSX Capture-OpenGL](https://drive.google.com/file/d/1g9AiCNheVCF22QK4ou9bYa9DqTF4c2i3/view?usp=sharing)

**Comment by legend800** [comment]
Motorstorm Apoc. is perfect (graphically) again. Thanks kd-11!

![2019-03-25 19_01_39-Window](https://user-images.githubusercontent.com/20983844/54966139-921e6200-4f30-11e9-8b19-fa1b18894315.png)

![2019-03-25 19_02_10-Window](https://user-images.githubusercontent.com/20983844/54966140-921e6200-4f30-11e9-9b0a-a28d724808d7.png)

**Comment by kd-11** [comment]
Quick update on bioshock - I know why it's not working too well but it needs quite a bit of work (old code was working because of a hack). Should be ready in the next day or so.

**Comment by stormymff** [comment]
what about Yakuza 4?

**Comment by MSuih** [comment]
Yakuza 4 is working fine on the kd's appveyor builds

**Comment by stormymff** [comment]
How to use it?

**Comment by MSuih** [comment]
1. https://ci.appveyor.com/project/kd-11/rpcs3/history
2. Click on the latest entry from fbo_fixes branch.
3. Switch to artifacts tab and download the largest file from there.

You could also ask questions like this at discord chat so that people following this regression issue woudn't get pointless notifications.

**Comment by stormymff** [comment]
My apologize sir. Thanks for the responds, this is my first time

**Comment by Xcedf** [comment]
@kd-11 just out of curiosity, since you're improving GPU Blit, is there any way to make GoW Ascension work with it, for now it only works with CPU Blit, with GPU Blit it has
E {RSX [0x026b69c]} RSX: nv406e::semaphore_acquire has timed out. semaphore_address=0x50300510
E {RSX [0x026c910]} RSX: nv406e::semaphore_acquire has timed out. semaphore_address=0x50300520
on older builds, before your tc rewriting i used a hack to make it work
disabling the line in texture_cache.h
cached_dest->reprotect(utils::protection::no, { mem_base, mem_length });
And it worked a lot faster than with CPU Blit, now it also works, except it cause the overall darkness ingame.

**Comment by kd-11** [comment]
Interesting, I'll look into it.

**Comment by kd-11** [comment]
Bioshock should now work correctly with builds from my repo. Check for any specific regressions. Only vulkan implementation is done, OpenGL in theory should be completely broken as format conversion is not yet implemented.
Uncharted 1 has a wierd glitch where it looks like motion blur is completely broken despite the fact that everything is being done accurately; this needs fixing before the commits are ready to PR.

**Comment by Xcedf** [comment]
Do not have Bishock to test but here's a portion of improvements of the last push
<details>
<summary>Baja: Edge of Control</summary>
The game still has some issues with skybox texture, but looks work a way better with GPU Blit now, besides maybe WCB+CPU Blit fix this, but work twice slower
before
![27](https://user-images.githubusercontent.com/7060586/55282496-d016ee00-5355-11e9-8578-ce365d607815.jpg)
after
![28](https://user-images.githubusercontent.com/7060586/55282499-d60ccf00-5355-11e9-8357-64933043e8ec.jpg)

</details>

<details>
<summary>Lolipop Chainsaw</summary>
The intro scene is not exactly totally fixed but also a way better
before
![12](https://user-images.githubusercontent.com/7060586/55282511-19673d80-5356-11e9-84b4-f29031530c67.jpg)
after
![13](https://user-images.githubusercontent.com/7060586/55282518-2126e200-5356-11e9-9d6f-8361078be8c8.jpg)

</details>

<details>
<summary>The Darkness Demo</summary>
Main Menu now looks fine, unfrotunately can't go ingame it freezes with Vulkan Driver Fail with GPU Blit
before
![31](https://user-images.githubusercontent.com/7060586/55282526-62b78d00-5356-11e9-9bb5-4b430dc8e382.jpg)
after
![30](https://user-images.githubusercontent.com/7060586/55282529-6e0ab880-5356-11e9-8be2-36350ad251c4.jpg)

</details>

**Comment by Emulator-Team-2** [comment]
@kd-11 
BioShock, Turok, Stranglehold are now fixed.

<details> 
<summary>BioShock</summary>

Before: 
![1](https://user-images.githubusercontent.com/38318848/55286861-c6b37300-53a1-11e9-9919-2e9745bcb391.png)
After:
![2](https://user-images.githubusercontent.com/38318848/55286866-da5ed980-53a1-11e9-9ea7-453713dc6b3d.png)

</details>

<details> 
<summary>Turok</summary>

Before: 
![3](https://user-images.githubusercontent.com/38318848/55286897-4e997d00-53a2-11e9-866f-f98ec9b0ad79.png)
After:
![4](https://user-images.githubusercontent.com/38318848/55286904-6244e380-53a2-11e9-9694-6c9d2d2b0221.png)

</details>

<details> 
<summary>Stranglehold</summary>

Before: 
![5](https://user-images.githubusercontent.com/38318848/55286919-94eedc00-53a2-11e9-8301-ed184d8f379d.png)
After:
![6](https://user-images.githubusercontent.com/38318848/55286925-9ddfad80-53a2-11e9-8ab7-b9a5c714a77e.png)

</details>

**Comment by kd-11** [comment]
Fixed uncharted 1 as well (cache lookup bug); only OpenGL implementation of format conversions remains now. Just check if anything regressed in my builds. A PR should be coming in the next 48h.

**Comment by kd-11** [comment]
Retest any remaining issues with https://github.com/RPCS3/rpcs3/pull/5785.

**Comment by MarioSonic2987** [comment]
MCLA and RE6 still look fine with #5785:
![image](https://user-images.githubusercontent.com/20027681/55417987-04ff8c80-5572-11e9-883e-ca3fd7df8543.png)
![image](https://user-images.githubusercontent.com/20027681/55418223-79d2c680-5572-11e9-80be-41c7fe362cdf.png)

**Comment by kd-11** [comment]
Closing as fixed by https://github.com/RPCS3/rpcs3/pull/5785
